crates/proto/src/errors/mod.rs

```diff
@@ -74,3 +77,65 @@ impl From<ConversionError> for tonic::Status {
         tonic::Status::invalid_argument(value.to_string())
     }
 }
+
+// GRPC ERROR TRAIT
+// ================================================================================================
+
+/// Common trait for all gRPC error enums
+pub trait GrpcError {
+    fn api_code(self) -> u8;
+    fn is_internal(&self) -> bool;
+
+    fn tonic_code(&self) -> tonic::Code {
+        if self.is_internal() {
+            tonic::Code::Internal
+        } else {
+            tonic::Code::InvalidArgument
+        }
+    }
+}
+
+// HELPER MACROS
+// ================================================================================================
+
+/// Implement the `GrpcError` trait for a given error type
+#[macro_export]
```

Comment:
Do we need to export this? Its internal to the crate iiuc. Though I always gets confused as to whether export means `pub` or just that its usable outside the module.

Reply:
The trait methods defined here are called from the `store` and `block-producer` crates, inside each domain error `Into<tonic::Status>` implementation, so we need to export it